### DISCUSSION TITLE
Creating a Popup window without standard window Controls

### DISCUSSION THREAD

**Discussion by SirZeeno** [AvaloniaUI/Avalonia#17256] [Q&A]
I am trying to make a popup window that pops up right when the program is launched and certain conditions are met, but I want the window to have its style and therefore its title bar that I am in full control of the title bar. Consequently, I am using fluent UI for Avalonia, but I am running into an issue that I can't seem to solve. 

The issue is that I want the window controls to be the ones I created, like a close button, but not the standard OS control. The close button is all I need and I want to be able to put my style on it, but haven't found anything concerning that, so I am stuck on how to solve this.

Any help is greatly appreciated :)

**Comment by thevortexcloud** [comment]
From memory: You can hide the frame and then extend your window out to the decorations.

The control catalogue looks like it has some samples for this:

https://github.com/AvaloniaUI/Avalonia/blob/7f2cd0723a1f5418fcda7e5a922948685c36e2e2/samples/ControlCatalog/Pages/WindowCustomizationsPage.xaml#L13

It is however a hint and may not be supported on all platforms.

**Reply by SirZeeno** [reply]
from what I can see, that checkbox is responsible for setting ExtendClientAreaToDecorationsHint in the main window to true or false, but I already use `ExtendClientAreaToDecorationsHint = true` which doesn't have the effect I am looking for. (for reference I am on Windows)

**Reply by thevortexcloud** [reply]
I _think_ you also need to set these:

```xaml
ExtendClientAreaToDecorationsHint="True"
ExtendClientAreaChromeHints="NoChrome"
ExtendClientAreaTitleBarHeightHint="0"
```

**Reply by SirZeeno** [reply]
nope, no dice.

![image](https://github.com/user-attachments/assets/17eaae9d-168b-4d7e-bad6-90fe589c7934)

I am already using these in my popup window, and none of these seem to do the trick
````
CanResize="False"
WindowStartupLocation="CenterOwner"
ExtendClientAreaToDecorationsHint="True"
ExtendClientAreaChromeHints="NoChrome"
ExtendClientAreaTitleBarHeightHint="0"
SystemDecorations="None"
ShowInTaskbar="False"
````

**Reply by thevortexcloud** [reply]
Hmm, I don't have any Windows devices so it's hard for me to test this. You can theoretically hack around it by using the Win32 APIs and the native window handle, but that's not ideal. 

https://stackoverflow.com/questions/2398746/removing-window-border

Someone else may have a better idea.

**Reply by SirZeeno** [reply]
I hope so. I want to remove the window controls, but I also don't want to get too hacky with something I will only need to use for one thing.

**Reply by SirZeeno** [reply]
Sooooo... I know where my problem was, i was using the AppWindow class extension from FluentAvalonia instead of Window from Avalonia, so there was no way for Avalonia to affect if the title bar was there or not.
